source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -50,324 +51,257 @@ def delete_journeys(responses, request):
-    if dp != 'indifferent':
-        filters.append(partial(filter_direct_path,
-                               is_debug=is_debug,
-                               dp=dp))
+def to_be_deleted(journey):
+    return 'to_delete' in journey.tags
 
-    composed_filter = ComposedFilter()
-    [composed_filter.add_filter(f) for f in filters]
 
-    return composed_filter.compose_filters()(get_qualified_journeys(responses))
+def mark_as_dead(journey, is_debug, *reasons):
+    journey.tags.append('to_delete')
+    if is_debug:
+        journey.tags.extend(('deleted_because_' + reason for reason in reasons))
 
 
-def final_filter_journeys(response_list, instance, request):
+@six.add_metaclass(abc.ABCMeta)
+class SingleJourneyFilter(object):
     """
-    Filter by side effect the list of pb responses's journeys
-    Final pass : we remove similar journeys (same lines and stop_points of change)
+    Interface to implement for filters applied to a single journey (no comparison)
     """
 
-    # for clarity purpose we build a temporary list
+    @abc.abstractmethod
+    def filter_func(self, journey):
+        """
+        :return: True if the journey is valid, False otherwise
+        """
+        pass
 
-    final_line_filter = get_or_default(request, '_final_line_filter', False)
-    if final_line_filter:
-        journeys = get_qualified_journeys(response_list)
-        journeys_pool = itertools.combinations(journeys, 2)
-        _filter_similar_line_journeys(journeys_pool, request)
+    @abc.abstractproperty
+    def message(self):
+        """
+        :attribute: a one-word, snake_case unicode to be used to explain why a journey is filtered
+        """
+        pass
 
-    journeys = get_qualified_journeys(response_list)
-    _filter_too_much_connections(journeys, instance, request)
 
-
-def _get_worst_similar(j1, j2, request):
-    """
-    Decide which is the worst journey between 2 similar journeys.
-
-    The choice is made on:
-     - asap
-     - duration
-     - more fallback
-     - more connection
-     - smaller value among min waiting duration
-     - more constrained fallback mode : all else being equal,
-            it's better to know that you can do it by bike for 'bike_in_pt' tag
-            (and traveler presumes he can do it walking too, as the practical case is 0s fallback)
+def filter_wrapper(filter_obj=None, is_debug=False):
     """
-    if request.get('clockwise', True):
-        if j1.arrival_date_time != j2.arrival_date_time:
-            return j1 if j1.arrival_date_time > j2.arrival_date_time else j2
-    else:
-        if j1.departure_date_time != j2.departure_date_time:
-            return j1 if j1.departure_date_time < j2.departure_date_time else j2
+    Wraps a SingleJourneyFilter instance to automatically deal with debug-mode, logging and tagging
 
-    if j1.duration != j2.duration:
-        return j1 if j1.duration > j2.duration else j2
+    We always tag the journey 'to_delete'
```

Comment:
always? It depends of the result of the filter, doesn't it? This sentence is confusing.